test/unit/realm.test.js

```diff
@@ -106,6 +106,23 @@ describe('=> Realm', () => {
       assert.equal(queries[0], 'SELECT 1');
     });
 
+    it('should be able to customize logger , even though logQuery will throw error', async () => {
+      const realm = new Realm({
+        port: process.env.MYSQL_PORT,
+        user: 'root',
+        database: 'leoric',
+        logger: {
+          logQuery(sql) {
+            throw new Error('asdf');
+          }
+        }
+      });
+      await realm.connect();
+      const res = await realm.driver.query('SELECT 1');
+      assert(res.rows);
+    });
```

Comment:
test/unit/drivers/abstract/index.test.js would be the better place to put this test case